src/js/extension/contentScript.js

```diff
@@ -0,0 +1,27 @@
+function injectScript() {
+    try {
+        const container = document.head || document.documentElement
+        const scriptTag = document.createElement('script')
+        scriptTag.setAttribute('async', 'false')
+        scriptTag.src = chrome.runtime.getURL('/js/extension/tonProvider.js');
```

Comment:
You need to wait until script will be loaded here. And then do runtime.connect otherwise "doMagic" logic will fail at race condition. Check my implementation: [contentscript](https://github.com/D2Phoenix/react-ton-wallet/blob/main/src/chrome/contentscript.ts)

Reply:
Thanks for spotting this. What do you think about [this way](https://github.com/toncenter/ton-wallet/pull/99/commits/3ee34387e7ea7f48b3efc34b5565e9cdf1cd7010) to fix that?

Reply:
@neverthirty it is looks correct

Reply:
implemented in master